cmd/planner-api/run.go

```diff
@@ -140,3 +141,39 @@ func newListener(address string) (net.Listener, error) {
 	}
 	return net.Listen("tcp", address)
 }
+
+func initializeIso(ctx context.Context, cfg *config.Config) error {
+	// Check if ISO already exists:
+	isoPath := util.GetEnv("MIGRATION_PLANNER_ISO_PATH", "rhcos-live.x86_64.iso")
+	if _, err := os.Stat(isoPath); err == nil {
+		return nil
+	}
+
+	out, err := os.Create(isoPath)
+	if err != nil {
+		return err
+	}
+	defer out.Close()
+
+	md := iso.NewDownloaderManager()
+
+	minio, err := iso.NewMinioDownloader(
+		iso.WithEndpoint(cfg.Service.S3.Endpoint),
+		iso.WithBucket(cfg.Service.S3.Bucket),
+		iso.WithAccessKey(cfg.Service.S3.AccessKey),
+		iso.WithSecretKey(cfg.Service.S3.SecretKey),
+		iso.WithImageName(cfg.Service.S3.IsoFileName),
+	)
+	if err == nil {
+		md.Register(minio)
+	}
+
+	// register the default downloader of the official RHCOS image.
+	md.Register(iso.NewRHCOSHttpDownloader())
+
+	if err := md.Download(ctx, out); err != nil {
```

Comment:
So if we download both, how do we decide which one to use? Should we download just one?

Reply:
we don't download both. It's trying one by one and stop on success

Reply:
Ah right, I see now.